File: libs/eave-stdlib-py/src/eave/stdlib/core_api/operations/github_repos.py
from typing import Unpack, Optional
import uuid

from eave.stdlib.api_types import BaseRequestBody, BaseResponseBody
from ... import requests
from eave.stdlib.core_api.models.github_repos import (
    GithubRepo,
    GithubRepoCreateInput,
    GithubRepoListInput,
    GithubReposDeleteInput,
    GithubRepoUpdateInput,
    GithubReposFeatureStateInput,
)
from . import CoreApiEndpoint, CoreApiEndpointConfiguration


class GetGithubReposRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-repos/query",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        repos: Optional[list[GithubRepoListInput]] = None

    class ResponseBody(BaseResponseBody):
        repos: list[GithubRepo]

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID | str,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=input,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )


class GetAllTeamsGithubReposRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(path="/_/github-repos/query", auth_required=False, team_id_required=False)

    class RequestBody(BaseRequestBody):
        query_params: GithubReposFeatureStateInput

    class ResponseBody(BaseResponseBody):
        repos: list[GithubRepo]

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=input,
            **kwargs,
        )


class FeatureStateGithubReposRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-repos/query/enabled",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        query_params: GithubReposFeatureStateInput

    class ResponseBody(BaseResponseBody):
        states_match: bool

    @classmethod
    async def perform(
        cls,
        team_id: uuid.UUID | str,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=None,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )


class CreateGithubRepoRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-repos/create",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        repo: GithubRepoCreateInput

    class ResponseBody(BaseResponseBody):
        repo: GithubRepo

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID | str,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=input,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )


class DeleteGithubReposRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-repos/delete",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        repos: list[GithubReposDeleteInput]

    class ResponseBody(BaseResponseBody):
        pass

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID | str,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=input,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )


class UpdateGithubReposRequest(CoreApiEndpoint):
    config = CoreApiEndpointConfiguration(
        path="/github-repos/update",
        auth_required=False,
    )

    class RequestBody(BaseRequestBody):
        repos: list[GithubRepoUpdateInput]

    class ResponseBody(BaseResponseBody):
        repos: list[GithubRepo]

    @classmethod
    async def perform(
        cls,
        input: RequestBody,
        team_id: uuid.UUID | str,
        account_id: Optional[uuid.UUID],
        access_token: Optional[str],
        **kwargs: Unpack[requests.CommonRequestArgs],
    ) -> ResponseBody:
        return await requests.make_request(
            config=cls.config,
            response_type=cls.ResponseBody,
            input=input,
            team_id=team_id,
            account_id=account_id,
            access_token=access_token,
            **kwargs,
        )

File: libs/eave-stdlib-ts/src/core-api/operations/github-repos.ts
import { makeRequest, RequestArgsAuthedRequest, RequestArgsTeamId } from "../../requests.js";
import { GithubRepo, GithubRepoCreateInput, GithubRepoListInput, GithubReposDeleteInput, GithubReposFeatureStateInput, GithubRepoUpdateInput } from "../models/github-repos.js";
import { CoreApiEndpointConfiguration } from "./shared.js";

export type GetGithubReposRequestBody = {
  repos?: Array<GithubRepoListInput>;
};

export type GetGithubReposResponseBody = {
  repos: Array<GithubRepo>;
};

export class GetGithubReposOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/github-repos/query" });
  static async perform(args: RequestArgsTeamId & { input: GetGithubReposRequestBody }): Promise<GetGithubReposResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <GetGithubReposResponseBody>await resp.json();
    return responseData;
  }
}
export type FeatureStateGithubReposRequestBody = {
  query_params: GithubReposFeatureStateInput;
};

export type FeatureStateGithubReposResponseBody = {
  states_match: boolean;
};

export class FeatureStateGithubReposOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/github-repos/query/enabled" });
  static async perform(args: RequestArgsTeamId & { input: FeatureStateGithubReposRequestBody }): Promise<FeatureStateGithubReposResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <FeatureStateGithubReposResponseBody>await resp.json();
    return responseData;
  }
}
export type CreateGithubRepoRequestBody = {
  repo: GithubRepoCreateInput;
};

export type CreateGithubRepoResponseBody = {
  repo: GithubRepo;
};

export class CreateGithubRepoOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/github-repos/create" });
  static async perform(args: RequestArgsTeamId & { input: CreateGithubRepoRequestBody }): Promise<CreateGithubRepoResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <CreateGithubRepoResponseBody>await resp.json();
    return responseData;
  }
}
export type DeleteGithubReposRequestBody = {
  repos: Array<GithubReposDeleteInput>;
};

export class DeleteGithubRepoOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/github-repos/delete" });
  static async perform(args: RequestArgsTeamId & { input: DeleteGithubReposRequestBody }): Promise<void> {
    await makeRequest({
      config: this.config,
      ...args,
    });
  }
}

export type UpdateGithubReposRequestBody = {
  repos: Array<GithubRepoUpdateInput>;
};

export type UpdateGithubReposResponseBody = {
  repos: Array<GithubRepo>;
};

export class UpdateGithubReposOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/github-repos/update" });
  static async perform(args: RequestArgsAuthedRequest & { input: UpdateGithubReposRequestBody }): Promise<UpdateGithubReposResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <UpdateGithubReposResponseBody>await resp.json();
    return responseData;
  }
}
